<template name="usage">
    <head>
        <title>Snaptron Usage</title>
    </head>

    <body>
    {{> navbar}}
    <div class="container">
        <div class="alert alert-dismissible alert-info">
            <button type="button" class="close" data-dismiss="alert">x</button>
            <h4>Under Construction</h4>
            <p>This website is under active development. Features may be broken, inconsistent, or missing. If you notice
                something, please <a href="https://github.com/ChristopherWilks/SnaptronUI/issues">submit an issue on
                    GitHub</a>.</p>
        </div>

        <h1>Overview</h1>
        <p>
            Snaptron allows you to explore exon-exon junctions from the Intropolis database (see <a
                href="/about">About</a>).
            <!-- All information is accessible, including junction evidence and sample information. 
	    Additionally, you can perform simple analysis on sets of junctions. -->
        </p>

        <h1>Basic Usage</h1>
        <h2>Viewing a Region</h2>
        <p>
            From the <a href="/">Home</a> page, create a new query by entering a region and submitting. 
	    Regions can be in one of the following formats.
	    Note: only junctions fully contained (inclusive) within the specified region as specified via either 
	    a gene name or coordinates are returned.

	    If gene region has never been loaded before or it's large or it contains more than a few hundred junctions,
	    it will take a while (seconds-to a few minutes) to load.
        </p>
        <ul>
            <li><strong>Gene Name</strong>: The name, such as <code>CD99</code> to lookup. The largest extent
                corresponding to the
                gene will be used.
            </li>
            <li><strong>Gene Extent</strong>: The format <code>chr#:x-y</code>, where <code>#</code> is the chromosome
                number, <code>x</code> is
                the starting position, and <code>y</code> is the ending position
            </li>
        </ul>
        <p>
	    Querying a gene which maps to multiple regions/chromosomes will result in a zoomed out view with little or no
	    junctions visible (though still displayed).  In this case choosing a gene model (described below) may help
	    identify region(s) to zoom into.  Double click a specific place on the viewing panel 
	    to zoom in toward that area to see junction arcs.
        </p>

        <h2>Filtering</h2>
        <p>
            Filters allow you to narrow down the junction set for a given region. Junctions can be filtered according to
            any number of attributes, and multiple filters can be used together. The first opportunity to filter is from
	    the home page. By using the <code>Filtering Options</code> panel, the initial result set can be filtered. This can
            be useful if the entered region(s) are large.

	    All filters are turned off by default, except the <code>Samples Count</code> filter 
	    which is set to be greater than 20.
	    This is to reduce the large number of sparsely occurring junctions from being loaded by default.
	    You may of course lower/turn off this filter if you find that certain splice sites don't have junctions.
	    However, be forewarned that turning it off or setting it to 1 will greatly increase the load time.

	    If you choose certain filters (e.g. <code>Intron Length</code> less than 3000) you may have to zoom in from the
	    default zoom level as junction arcs may not be visible due to their small size.
        </p>
        <p>
            Filters can be modified by returning to the original search page, changing the filter setting, and resubmitting.
	    (click on the <code>Snaptron</code> link in upper left corner to return to the original search page).
            Filters are applied in the order they appear, and a
            junction must pass all filters for it to appear.
        </p>

	<h2>Gene Model (dropdown)</h2>
	<p>
		Most users will want to select a gene model (usually the one with
		the most exons) from the <code>Select Gene Model</code> drop down.  The vertical
		blue boxes are exons and the green line is the coding sequence
		extent (while it goes through the introns it's understood to just
		represent start/end of CDS of the gene model).
	</p>


	<h2>Color By (dropdown)</h2>
	<p>
		The <code>Color by</code> dropdown is useful to see which splice junctions are more
		supported (in more samples) than others.  A suggested option is <code>log(samples_count)</code>.
		This may help the user to find which of the junctions are more likely to be real
		since a lot of the junctions are almost certainly spurious.
	</p>

        <h2>Junction Data Table</h2>
	<p>
		The raw junction data itself is available at the bottom of the
		viewing page in the <code>All Junction Data</code> arrow though not all the
		columns are shown by default.

		The raw junction table can be used to quickly find the exact junction(s)
		by searching the page for a specific coordinate.
	</p>

    </div>
    </body>
</template>
